### PR TITLE
rpcdaemon: improve security in eth_getLogs code

### PR DESCRIPTION
**Pull request by lupin012**

### CODE REVIEW

File: silkworm/silkrpc/json/types.cpp [hunk 1]
```diff
@@ -31,22 +32,28 @@ namespace silkworm::rpc {
...
+    *dest = '\0';
 }
 
-std::size_t to_hex_no_leading_zeros(char* hex_bytes, silkworm::ByteView bytes) {
+std::size_t to_hex_no_leading_zeros(std::span<char> hex_bytes, silkworm::ByteView bytes) {
     static const char* kHexDigits{"0123456789abcdef"};
-
-    char* dest{&hex_bytes[0]};
     size_t len = bytes.length();
+    if (len * 2 + 2 > hex_bytes.size()) {
```
Comment (canepat) [inline silkworm/silkrpc/json/types.cpp:53]: This should be `len * 2 + 2 + 1` to count also for the terminator character '\0'

File: silkworm/silkrpc/json/types.cpp [hunk 2]
```diff
@@ -70,24 +77,25 @@ std::size_t to_hex_no_leading_zeros(char* hex_bytes, silkworm::ByteView bytes) {
...
 }
 
-std::size_t to_quantity(char* quantity_hex_bytes, intx::uint256 number) {
+std::size_t to_quantity(std::span<char> quantity_hex_bytes, intx::uint256 number) {
     if (number == 0) {
         quantity_hex_bytes[0] = '0';
         quantity_hex_bytes[1] = 'x';
         quantity_hex_bytes[2] = '0';
+        quantity_hex_bytes[3] = '\0';
         return 3;
```
Comment (canepat) [inline silkworm/silkrpc/json/types.cpp:99]: This should be `4` to count also for the terminator character '\0'

File: silkworm/silkrpc/json/types.cpp [hunk 3]
```diff
@@ -855,20 +864,25 @@ void make_glaze_json_error(std::string& reply, uint32_t id, const int code, cons
     glz::write_json(glaze_json_error, reply);
 }
 
+// static constexpr auto addressSize = 64;
+// static constexpr auto hashSize = 128;
+// static constexpr auto int64Size = 32;
+// static constexpr auto dataSize = 4096;
+
```
Comment (canepat) [inline silkworm/silkrpc/json/types.cpp:871]: Remove commented code

**Review by canepat — CHANGES_REQUESTED** [review]

File: silkworm/silkrpc/commands/eth_api.cpp [hunk 4]
```diff
@@ -1553,59 +1553,21 @@ boost::asio::awaitable<void> EthereumRpcApi::handle_eth_uninstall_filter(const n
 }
 
 // https://eth.wiki/json-rpc/API#eth_getlogs
-boost::asio::awaitable<void> EthereumRpcApi::handle_eth_get_logs(const nlohmann::json& request, nlohmann::json& reply) {
-    auto params = request["params"];
-    if (params.size() != 1) {
-        auto error_msg = "invalid eth_getLogs params: " + params.dump();
-        SILKRPC_ERROR << error_msg << "\n";
-        reply = make_json_error(request["id"], 100, error_msg);
+boost::asio::awaitable<void> EthereumRpcApi::handle_eth_glaze_get_logs(const nlohmann::json& request, std::string& reply) {
```
[the remark this answers is not in the document]
Comment (lupin012) [reply silkworm/silkrpc/commands/eth_api.cpp]: ok

File: silkworm/silkrpc/commands/eth_api.cpp [hunk 5]
```diff
@@ -1553,59 +1553,21 @@ boost::asio::awaitable<void> EthereumRpcApi::handle_eth_uninstall_filter(const n
...
+    if (!request.contains("params")) {
+        auto error_msg = "missing value for required argument 0";
+        SILKRPC_ERROR << error_msg << request.dump() << "\n";
+        make_glaze_json_error(reply, request["id"], -32602, error_msg);
         co_return;
     }
```
[the remark this answers is not in the document]
Comment (lupin012) [reply silkworm/silkrpc/commands/eth_api.cpp:1562]: We should add this test in many apis

File: silkworm/silkrpc/json/types.cpp [hunk 6]
```diff
@@ -31,22 +32,28 @@ namespace silkworm::rpc {
 
 using evmc::literals::operator""_address;
 
-void to_hex(char* hex_bytes, silkworm::ByteView bytes) {
+void to_hex(std::span<char> hex_bytes, silkworm::ByteView bytes) {
     static const char* kHexDigits{"0123456789abcdef"};
-    char* dest{&hex_bytes[0]};
+    if (bytes.size() * 2 + 2 > hex_bytes.size()) {
```
[the remark this answers is not in the document]
Comment (lupin012) [reply silkworm/silkrpc/json/types.cpp]: yes

File: silkworm/silkrpc/json/types.cpp (hunk 1 above)
Comment (lupin012) [reply silkworm/silkrpc/json/types.cpp]: yes

File: silkworm/silkrpc/json/types.cpp (hunk 2 above)
Comment (lupin012) [reply silkworm/silkrpc/json/types.cpp:99]: no the null is not counted

File: silkworm/silkrpc/json/types.cpp (hunk 3 above)
Comment (lupin012) [reply silkworm/silkrpc/json/types.cpp]: yes

File: silkworm/silkrpc/json/types.cpp (hunk 2 above)
Comment (canepat) [reply silkworm/silkrpc/json/types.cpp]: TBH it all depends on the meaning of the return type. If we agree to make it `void` as returned value seems not to be used anywhere, we can simply forget.

**Review by canepat — APPROVED** [review]

**Pull request merged**